maisim/maisim.Game/Utils/TestUtil.cs

```diff
@@ -1,172 +1,283 @@
+                DifficultyLevel = GetRandomDifficultyLevel(),
+                DifficultyRating = random.Next(1, 10),
+                NoteDesigner = GetRandomName()
+            };
+        }
+
+        /// <summary>
+        /// Return a <see cref="Score"/> with complete random data.
+        ///
+        /// You can put <see cref="Beatmap"/> as parameter if you want to use a specific <see cref="Beatmap"/> as property for this <see cref="Score"/>.
+        /// </summary>
+        public static Score CreateMockScore(Beatmap beatmap = null)
+        {
+            if (beatmap == null)
+            {
+                beatmap = CreateMockBeatmap();
+            }
+
+            float accuracy = random.NextFloatInRange(1, 100);
+
+            return new Score()
+            {
+                Beatmap = beatmap,
+                Tap = random.NextInRange(1, 200),
+                Hold = random.NextInRange(1, 200),
+                Slide = random.NextInRange(1, 200),
+                Touch = random.NextInRange(1, 200),
+                Accuracy = accuracy,
+                Rank = ScoreProcessor.CalculateRank(accuracy),
+                Combo = random.NextInRange(1, 500),
+            };
+        }
+
+        /// <summary>
+        /// Get the audio file path that's live in game's resources for testing.
+        /// When using this please set <see cref="BeatmapSet"/>'s `UseLocalPath` to `true`.
+        /// </summary>
+        /// <param name="trackMetadata">The target <see cref="TrackMetadata"/>.</param>
+        /// <returns>Path using in game's resource</returns>
+        public static string GetBeatmapSetAudioPath(TrackMetadata trackMetadata)
+        {
+            string title = trackMetadata.Title;
+
+            Logger.Log("GetBeatmapSetAudioPath: " + title);
+
+            switch (title)
+            {
+                case "Diamond City Lights":
+                    return "Test/diamond-city-lights.mp3";
+                case "only my railgun":
+                    return "Test/only-my-railgun.m4a";
+                case "RAISE MY SWORD":
+                    return "Test/raise-my-sword.mp3";
+                case "Sukino Skill":
+                    return "Test/sukino-skill.mp3";
+                case "Tenkai e no Kippu":
+                    return "Test/tenkai-e-no-kippu.mp3";
+                case "ReI":
+                    return "Test/rei.mp3";
+                default:
+                    return "";
+            }
+        }
+
+        /// <summary>
+        /// Get the preview time of the audio file for target <see cref="TrackMetadata"/>.
+        /// </summary>
+        /// <param name="trackMetadata">The target <see cref="TrackMetadata"/></param>
+        /// <returns>Preview time</returns>
+        public static int GetBeatmapSetPreviewTime(TrackMetadata trackMetadata)
+        {
+            string title = trackMetadata.Title;
+
+            switch (title)

```

Comment:
String matching for this is especially fragile, you may want to
move these informations inside the beatmap info.

Reply:
Since I fix on this and I see that every function in `TestUtil` are depend on the string match (the Japanese language song too) so I refactor it to using the enum instead. So it's easier to use `TestUtil` for testing (or selecting specific song) now.